bundle/apps/upload_config.go

```diff
@@ -47,25 +45,23 @@ func (u *uploadConfig) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnos
 			errGroup.Go(func() error {
 				err := f.Write(ctx, path.Join(appPath, "app.yml"), buf, filer.OverwriteIfExists)
 				if err != nil {
-					mu.Lock()
-					diags = append(diags, diag.Diagnostic{
+					diags.Append(diag.Diagnostic{
 						Severity:  diag.Error,
 						Summary:   "Failed to save config",
 						Detail:    fmt.Sprintf("Failed to write %s file: %s", path.Join(app.SourceCodePath, "app.yml"), err),
 						Locations: b.Config.GetLocations("resources.apps." + key),
 					})
-					mu.Unlock()
 				}
 				return nil
 			})
 		}
 	}
 
 	if err := errGroup.Wait(); err != nil {
-		return diags.Extend(diag.FromErr(err))
+		diags.Error(err)
```

Comment:
Missing a return.

Reply:
It's not missing -- diags.Error appends it and it is returned below.

Reply:
Ah, I see. It makes for pretty awkward control flow, IMO.

Either the `Error` function should be renamed to something active, e.g. `AddError` or `ExtendError`, or it should return in place, like `return diags.Error(err).Diags` (also not great).

What I worry about is replacing `diag.Diagnostics` with one of these and expecting the same outcome. The value for `diag.Diagnostics` can never be updated in place because it is appended to.

Reply:
That's true, but not if the slice is appended to beyond capacity.

Reply:
e66235c4c8c14e2fa2cafa4ad9adc4256649dc92